youtube_dl/extractor/facebook.py

```diff
@@ -498,25 +494,31 @@ def _lowercase_escape(s):
 
 
 
-    def get_from_new_ui(self, webpage, tahoe_data, url):
+    def get_from_new_ui(self, webpage, tahoe_data, video_id):
 
         video_title = self._search_regex(r'"headline":"(.+?")', webpage, 'title', fatal=False)
         if not video_title:
             video_title = self._search_regex(r'"pageTitle">(.+?)<', webpage, 'title', fatal=False)
         if not video_title:
             video_title = self._extract_video_title(webpage, tahoe_data, video_id)
+
         comments_count = parse_count(self._search_regex(r'"commentCount":(.+?,)', webpage, 'comments_count', fatal=False))
+        if comments_count is None:
+            comments_count = parse_count(self._search_regex(r'"commentcount":(.+?,)', tahoe_data.secondary, 'comments_count', fatal=False))
+        if comments_count is None:
+            comments_count = parse_count(self._extract_comments_count(webpage, tahoe_data))
+
         likes = parse_count(self._extract_likes(webpage, tahoe_data))
 
-        timestamp = self._search_regex(r'"datePublished":"(.+?)"', webpage, 'timestamp')
+        timestamp = self._search_regex(r'"datePublished":"(.+?)"', webpage, 'timestamp', fatal=False)
```

Comment:
can we move all the logic around timestamp into the _resolve_timestamp function?